feg/gateway/services/n7_n40_proxy/notify/notify_handler.go

```diff
@@ -0,0 +1,206 @@
+		return fmt.Errorf("internal server error")
+	}
+	defer client.Close()
+
+	var smUpdateNotify n7_client.SmPolicyNotification
+	err = ctx.Bind(&smUpdateNotify)
+	if err != nil {
+		err = fmt.Errorf("invalid SmPolicyNotification received: %s", err)
+		glog.Errorf("postSmPolicyUpdateNotification: %s", err)
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+	sessionId, imsi, err := getSessionIdAndIMSI(ctx.Param(EncodedSessionId))
+	if err != nil {
+		glog.Errorf("postSmPolicyUpdateNotification unable to fetch session-id for UpdateNotify - %s", err)
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+
+	policyReauthProto := n7.GetPolicyReauthRequestProto(sessionId, imsi, smUpdateNotify.SmPolicyDecision)
+	ans, err := client.PolicyReAuth(context.Background(), policyReauthProto)
+	if err != nil {
+		glog.Errorf("Error relaying N7 policy reauth request to gateway: %s", err)
+		return fmt.Errorf("internal server error")
+	}
+	partialSuccessReport := n7.GetPartialSuccessReportN7(ans)
+	if partialSuccessReport != nil {
+		ctx.JSON(http.StatusOK, partialSuccessReport)
+	}
+	return ctx.NoContent(http.StatusOK)
+}
+
+func (handler *NotificationHandler) postSmPolicyTerminateNotification(ctx echo.Context) error {
+	client, err := relay.GetAbortSessionResponderClient(handler.cloudRegistry)
+	if err != nil {
+		glog.Errorf("postSmPolicyTerminateNotification failed to get AbortSessionResponderClient: %s", err)
+		return fmt.Errorf("internal server error")
+	}
+	defer client.Close()
+
+	var smTermNotify n7_client.TerminationNotification
+	err = ctx.Bind(&smTermNotify)
+	if err != nil {
+		err = fmt.Errorf("invalid TerminateNotification received: %s", err)
+		glog.Errorf("postSmPolicyTerminateNotification: %s", err)
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+
+	sessionId, imsi, err := getSessionIdAndIMSI(ctx.Param(EncodedSessionId))
+	if err != nil {
+		glog.Errorf("postSmPolicyTerminateNotification unable to fetch session-id for TerminateNotify - %s", err)
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+
+	ans, err := client.AbortSession(context.Background(), &protos.AbortSessionRequest{
+		UserName:  imsi,
+		SessionId: sessionId,
+	})
+	if err != nil {
+		glog.Errorf("postSmPolicyTerminateNotification error relaying ASR to gateway: %s", err)
+		return ctx.NoContent(http.StatusNoContent)
+	}
+
+	switch ans.Code {
+	case protos.AbortSessionResult_SESSION_NOT_FOUND:
+		err = ctx.NoContent(http.StatusNotFound)
+	case protos.AbortSessionResult_USER_NOT_FOUND:
+		err = ctx.NoContent(http.StatusNotFound)
+	case protos.AbortSessionResult_GATEWAY_NOT_FOUND:
+		err = ctx.NoContent(http.StatusInternalServerError)
+	case protos.AbortSessionResult_RADIUS_SERVER_ERROR:
+		err = ctx.NoContent(http.StatusInternalServerError)
+	case protos.AbortSessionResult_SESSION_REMOVED:
+		err = ctx.NoContent(http.StatusNoContent)
+	default:
+		err = ctx.NoContent(http.StatusNoContent)
+	}
+	return err
+}
```

Comment:
can we be consistent on how we build the errors we return

Shouldn't those errors contain the string of the error itself too? (i dont remember if the string can be embedded on http error)

If you have a logic why you are adding the errors this way, please share it

Reply:
can we just return in the switch?
```
	switch ans.Code {
	case protos.AbortSessionResult_SESSION_NOT_FOUND:
		return echo.NewHTTPError(http.StatusNotFound, "Session not found")
	case protos.AbortSessionResult_USER_NOT_FOUND:
...

```
Simpler to read